run-tester.ps1

```diff
@@ -1,18 +1,18 @@
 
-[string]$nvdaVersion = "2023.3"
+[string]$nvdaVersion = [System.IO.Path]::GetFileNameWithoutExtension($env:NVDA_PORTABLE_ZIP)
```

Comment:
In this particular case, my opinion differs quite a bit from what I would probably do in bash land/or for script intended to be used >anywhere< else. Because we are launching this `run-tester.ps1` directly from the github powersell - I'm not as comfortable with how ps1 scripts pass arguments (not even sure if $1 - $9 could be relied on for example)

Even if we were to convert to taking in the positional arguments, or creating a [cmdlet](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_scripts?view=powershell-7.4) calling the script from the github runner would still need to push these values into environment to be able to use them (I think, taking from the example of the webrequest:)

```yaml
      - name: Log job state RUNNING
        shell: powershell
        if: inputs.status_url
        run: |
          $headerbits = $env:ARIA_AT_CALLBACK_HEADER -split ":\s*", 2
          $headers = @{$headerbits[0]=$headerbits[1]; "Content-Type" = "application/json"}
          $body = @{'status'='RUNNING'; 'externalLogsUrl'="$env:GITHUB_SERVER_URL/$env:GITHUB_REPOSITORY/actions/runs/$env:GITHUB_RUN_ID"; 'work_dir'="$env:ARIA_AT_WORK_DIR"} | ConvertTo-JSON
          Invoke-WebRequest $env:ARIA_AT_STATUS_URL -Headers $headers -Method 'POST' -Body $body
```

Future-proofing a bit here makes me want to perhaps pull all the environment vairables at the top of the ps1 script first. so it's quite obvious upon opening the file what it needs from the environment.

Happy to continue this discussion elsewhere also, just putting the first of the replies here.

Reply:
Makes sense. Thanks!